tests/test_patrol_snapshot.py

```
@@ -0,0 +1,96 @@
+
+
+@pytest.mark.asyncio
+async def test_snapshot_assistant_picks_earliest_expiry():
+    """多条助理防抖并存时，应选 created_at + delay 最早到期的那条，
+    而不是最后插入的。"""
+    dm = make_manager()
+    # 先插一条长等待（created_at 更早，但到期晚）
+    dm._assistant[("chat:g:1", "u1")] = make_record(
+        "assistant", "chat:g:1", 100.0, sender_id="u1"
+    )
+    # 后插一条短等待（created_at 更晚，但到期更早）
```

Ruff (0.16.1)

[warning] Comment contains ambiguous `（` (FULLWIDTH LEFT PARENTHESIS). Did you mean `(` (LEFT PARENTHESIS)?

(RUF003)